docs/user-guide/petset.md

```diff
@@ -0,0 +1,281 @@
+
+Saving this config into `petset.yaml` and submitting it to a Kubernetes cluster should creae the defined Pet Set and Pets it manages:
+
+```shell
+$ kubectl create -f petset.yaml
+service "nginx" created
+petset "nginx" deleted
+```
+
+## Pet Identity
+
+The identity of a Pet sticks to it, regardless of which node it's (re) scheduled on. We can examine the identity of the pets we just created.
+
+### Ordinal index
+
+you should see 2 pods with predictable names formatted thus: `$(petset name)-$(ordinal index assigned by petset controller)`
+
+```shell
+$ kubectl get po
+NAME      READY     STATUS    RESTARTS   AGE
+web-0     1/1       Running   0          10m
+web-1     1/1       Running   0          10m
+```
+
+### Stable storage
+
+2 persistent volumes, one per pod. This is auto created by the Pet Set based on the `volumeTemplate` field
+
+```shell
+$ kubectl get pv
+NAME                                       CAPACITY   ACCESSMODES   STATUS    CLAIM               REASON    AGE
+pvc-90234946-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-0             11m
+pvc-902733c2-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-1             11m
+```
+
+### Network identity
+
+The network identity has 2 parts. First, we created a headless Service that controls the domain within which we create Pets. The domain managed by this Service takes the form: `$(service name).$(namespace).svc.cluster.local`, where "cluster.local" is the [cluster domain](http://releases.k8s.io/{{page.githubbranch}}/build/kube-dns/README.md#how-do-i-configure-it). As each pet is created, it gets a matching DNS subdomain, taking the form: `$(petname).$(governing service domain)`, where the governing service is defined by the `serviceName` field on the Pet Set.
+
+Cluster Domain | Service (ns/name) | Pet Set (ns/name) | Pet Set Domain | Pet DNS | Pet Hostname |
+-------------- | ----------------- | ----------------- | -------------- | ------- | ------------ |
+ cluster.local | default/nginx     | default/web       | nginx.default.svc.cluster.local | web-{0..N-1}.nginx.default.svc.cluster.local | web-{0..N-1} |
+ cluster.local | foo/nginx         | foo/web           | nginx.foo.svc.cluster.local     | web-{0..N-1}.nginx.foo.svc.cluster.local     | web-{0..N-1} |
+ kube.local    | foo/nginx         | foo/web           | nginx.foo.svc.kube.local        | web-{0..N-1}.nginx.foo.svc.kube.local        | web-{0..N-1} |
+
+Lets verify this assertion with a simple test.
+
+```shell
+$ kubectl get svc
+NAME          CLUSTER-IP     EXTERNAL-IP       PORT(S)   AGE
+nginx         None           <none>            80/TCP    12m
+```
+
+The containers are running nginx webservers, which by default will look for an index.html file in `/usr/share/nginx/html/index.html`. That directory is backed by a PersistentVolume created by the Pet Set. So lets write our hostname there (remember the Pet Set gives us a stable hostname):
+
```

Comment:
It is cool how you illustrate so many features with one command.  I wonder if some people might learn better if you did it in more steps.  For example:

1. show that the hostname inside the pod-0 is pod-0, with command: `for i in 0 1; do kubectl exec web-$i -- sh -c 'hostname'; done`
2. show that cluster DNS has the DNS addresses by making a third, unrelated pod, and nslookup-ing the expected DNS address.
3. show that when I write a file to a file within the pod's local volume, nginx serves it.
4. show that when you delete the pods, the data is still there.